nevergrad/optimization/optimizerlib.py

```diff
@@ -2407,7 +2407,7 @@ def optim(self) -> base.Optimizer:
         if self._optim is None:
             self._optim = self._select_optimizer_cls()(self.parametrization, self.budget, self.num_workers)
             self._optim = self._optim if not isinstance(self._optim, NGOptBase) else self._optim.optim
-            logger.debug("%s selected %s optimizer.", *(x.name for x in (self, self._optim)))
+        logger.debug("%s selected %s optimizer.", *(x.name for x in (self, self._optim)))
```

Comment:
This should only be called at initialization
```suggestion
            logger.debug("%s selected %s optimizer.", *(x.name for x in (self, self._optim)))
```